staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1646,6 +1646,32 @@ func (c *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID str
 	return false, nil
 }
 
+// InstanceMetadataByProviderID returns metadata of the specified instance.
+func (c *Cloud) InstanceMetadataByProviderID(ctx context.Context, providerID string) (*cloudprovider.InstanceMetadata, error) {
+	instanceID, err := KubernetesInstanceID(providerID).MapToAWSInstanceID()
+	if err != nil {
+		return nil, err
+	}
+
+	instance, err := describeInstance(c.ec2, instanceID)
+	if err != nil {
+		return nil, err
+	}
+	if *instance.State.Name == ec2.InstanceStateNameTerminated {
```

Comment:
It seems this check isn't there neither in NodeAddressesByProviderID nor in InstanceTypeByProviderID

Reply:
It's a logic of [InstanceExistsByProviderID](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go#L1605).
I had thought it's useful to judge whether an instance exists.

Reply:
Yeah - I've seen that. But it wasn't used in NodeAddressesByProviderID.

So to make this a no-op change (which is really a goal now), let's remove it for now.
Same below.

You can add a TODO to validate if we don't want to add it.

Reply:
all done.